### PR TITLE
[SPARK-31234][SQL][FOLLOW-UP] ResetCommand should not affect static SQL Configuration

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?
This PR is the follow-up PR of https://github.com/apache/spark/pull/28003

- add a migration guide
- add an end-to-end test case.

### Why are the changes needed?
The original PR made the major behavior change in the user-facing RESET command. 

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Added a new end-to-end test

### CODE REVIEW

File: docs/core-migration-guide.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ license: |
 ## Upgrading from Core 2.4 to 3.0
 
 - The `org.apache.spark.ExecutorPlugin` interface and related configuration has been replaced with
-  `org.apache.spark.plugin.SparkPlugin`, which adds new functionality. Plugins using the old
+  `org.apache.spark.api.plugin.SparkPlugin`, which adds new functionality. Plugins using the old
```
Comment (gatorsmile) [inline docs/core-migration-guide.md:28]: This is a typo fix.

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -216,6 +216,10 @@ license: |
 
     * The decimal string representation can be different between Hive 1.2 and Hive 2.3 when using `TRANSFORM` operator in SQL for script transformation, which depends on hive's behavior. In Hive 1.2, the string representation omits trailing zeroes. But in Hive 2.3, it is always padded to 18 digits with trailing zeroes if necessary.
 
+## Upgrading from Spark SQL 2.4.5 to 2.4.6
+
+  - In Spark 2.4.6, the `RESET` command does not reset the static SQL configuration values to the default. It only clears the runtime SQL configuration values.
```
Comment (gatorsmile) [inline docs/sql-migration-guide.md:221]: This assume https://github.com/apache/spark/pull/28262 will be merged to 2.4.6

Comment (dongjoon-hyun) [reply docs/sql-migration-guide.md:221]: Yes. #28262 is merged a few minutes ago.

**Comment by maropu** [comment]
Yea, the fix looks fine if the relevant test failure fixed.

**Review by yaooqinn — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 3]
```diff
@@ -163,9 +163,9 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach {
       .getOrCreate()
 
     assert(session.sessionState.conf.getConfString("spark.app.name") === "test-app-SPARK-31234")
-    assert(session.sessionState.conf.getConf(GLOBAL_TEMP_DATABASE) === "globalTempDB-SPARK-31234")
+    assert(session.sessionState.conf.getConf(GLOBAL_TEMP_DATABASE) === "globaltempdb-spark-31234")
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:166]: This difference between Spark 2.4 and Spark 3.0 is caused by https://github.com/apache/spark/pull/24979/

**Comment by gatorsmile** [comment]
cc @cloud-fan

**Comment by cloud-fan** [comment]
LGTM

**Comment by maropu** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merged to master and 3.0

**Pull request closed without merging**